kernel/src/process.rs

```diff
@@ -84,6 +69,130 @@ pub unsafe fn load_processes(
     }
 }
 
+/// This trait is implemented by process structs.
+pub trait ProcessType {
+    /// Queue a callback for the process. This will be added to a per-process
+    /// buffer and passed to the process by the scheduler.
+    fn schedule(&self, callback: FunctionCall) -> bool;
+
+    /// Queue an IPC operation for this process.
+    fn schedule_ipc(&self, from: AppId, cb_type: IPCType);
+
+    /// Remove the scheduled operation from the front of the queue.
+    fn unschedule(&self) -> Option<Task>;
+
+    /// Returns the current state of the process. Common states are "running" or
+    /// "yielded".
+    fn current_state(&self) -> State;
+
+    /// Move this process from the running state to the yield state.
+    fn yield_state(&self);
+
+    /// Put this process in the fault state. This will trigger the
+    /// `FaultResponse` for this process to occur.
+    unsafe fn fault_state(&self);
+
+    /// Get the name of the process. Used for IPC.
+    fn get_package_name(&self) -> &[u8];
```

Comment:
Feels weird to have the method `get_package_name` and the comment `Get the name of the process`. Is there a reason we aren't just calling this the process name?

Reply:
I don't think so

Reply:
It's called package name everywhere, but colloquially it is easier to call it "process name". I have changed process.rs to use `process_name` instead.